frontend/src/lib/search-utils.ts

```diff
@@ -22,32 +22,61 @@ export const handleStandardComparison = (valueOne: string, valueTwo: string, ope
   }
 }
 
-// for a given displayVersion string there is a distribution value a ' ' and a semver value, here we divide them and take the semver
-export const handleSemverOperatorComparison = (versionOne: string, versionTwo: string, operator: SearchOperator) => {
-  // Semver coerces the version to a valid semver version if possible, otherwise it returns the original value
-  const coercedVersionOne = semver.valid(semver.coerce(versionOne)) ?? versionOne
-  const coercedVersionTwo = semver.valid(semver.coerce(versionTwo)) ?? versionTwo
+export const handleSemverOperatorComparison = (
+  versionToCompare: string,
+  inputVersion: string,
+  operator: SearchOperator
+) => {
+  const coerceVersion = (version: string) => semver.valid(semver.coerce(version)) ?? version
```

Comment:
Does using `semver.valid` accomplish anything here? I think `semver.coerce` returns either a valid SemVer or null. Perhaps we could just do `const coercedInputVersion = semver.coerce(inputVersion)` and then just check for a value in the later conditions, rather than checking the result of `semver.coerce` again.